Don't ignore the help_type parameter

Sharpie opened the pull request:
When helpr is loaded, it provides a new default method for:

```
print.help_files_with_topic
```

Currently, this method always serves HTML help, even if a user provides a
different argument to the help function such as `help(..., help_type =
'text')`.

This patch fixes this behavior by calling back to
`utils:::print.help_files_with_topic` if the requested help type was something
other than 'html'.

hadley commented:
Good idea, but I think it would be better to return early, right after the type is determined, and skip all of the unnecessary code.

Sharpie committed:
Don't ignore the help_type parameter

When helpr is loaded, it provides a new default method for:

print.help_files_with_topic

Currently, this method always serves HTML help, even if a user provides a
different argument to the help function such as `help(..., help_type =
'text')`.

This patch fixes this behavior by calling back to
`utils:::print.help_files_with_topic` if the requested help type was something
other than 'html'.

Sharpie commented:
Commit amended.
